server/integration/endpoint/src/main/java/org/infinispan/server/endpoint/subsystem/EndpointSubsystemReader_9_0.java

```diff
@@ -303,6 +340,46 @@ private void parseWebSocketConnector(XMLExtendedStreamReader reader, PathAddress
       operations.add(connector);
    }
 
+   private void parseRouterConnector(XMLExtendedStreamReader reader, PathAddress subsystemAddress,
+                                     List<ModelNode> operations) throws XMLStreamException {
+      ModelNode connector = Util.getEmptyOperation(ADD, null);
+      String name = ModelKeys.ROUTER_CONNECTOR;
+
+      for (int i = 0; i < reader.getAttributeCount(); i++) {
+         ParseUtils.requireNoNamespaceAttribute(reader, i);
+         String value = reader.getAttributeValue(i);
+         Attribute attribute = Attribute.forName(reader.getAttributeLocalName(i));
+         name = parseRouterConnectorAttributes(reader, connector, name, i, value, attribute);
+      }
+
+      PathAddress connectorAddress = subsystemAddress.append(PathElement.pathElement(ModelKeys.ROUTER_CONNECTOR, name));
+      connector.get(OP_ADDR).set(connectorAddress.toModelNode());
+      operations.add(connector);
+
+      //Since nextTag() moves the pointer, we need to make sure we won't move too far
+      boolean skipTagCheckAtTheEnd = reader.hasNext();
```

Comment:
Is this really necessary? There seem to be `reader.hasNext()` checks everywhere.